hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/HoodieTimelineArchiver.java

```diff
@@ -491,14 +503,33 @@ private Stream<HoodieInstant> getInstantsToArchive() {
     // active timeline. This is required by metadata table,
     // see HoodieTableMetadataUtil#processRollbackMetadata for details.
     if (HoodieTableMetadata.isMetadataTable(config.getBasePath())) {
-      HoodieTableMetaClient dataMetaClient = HoodieTableMetaClient.builder()
-          .setBasePath(HoodieTableMetadata.getDatasetBasePath(config.getBasePath()))
+      HoodieTableMetaClient metadataTableMetaClient = HoodieTableMetaClient.builder()
```

Comment:
This is the meta client for the data table, not the metadata table.

Reply:
you're right. Not sure why `HoodieTableMetadata` has two methods for the same purpose: `getDatasetBasePath` and `getDataTableBasePathFromMetadataTable`